hikari/impl/event_manager_base.py

```diff
@@ -362,15 +371,29 @@ def __init__(
                 event_name = name[3:]
                 if isinstance(member, _FilteredMethodT):
                     caching = (member.__cache_components__ & cache_components) != 0
-                    self._consumers[event_name] = _Consumer(member, member.__event_types__, caching)
+
+                    self._consumers[event_name] = _Consumer(member, member.__events_bitmask__, caching)
 
                 else:
-                    self._consumers[event_name] = _Consumer(
-                        member, undefined.UNDEFINED, cache_components != cache_components.NONE
-                    )
+                    self._consumers[event_name] = _Consumer(member, -1, cache_components != cache_components.NONE)
 
-    def _clear_enabled_cache(self) -> None:
-        self._enabled_consumers_cache = {}
+    def _increment_listener_group_count(
+        self,
+        event_type: typing.Type[base_events.Event],
+        count: typing.Literal[-1, 1],
```

Comment:
```suggestion
        self,
        event_type: typing.Type[base_events.Event],
        count: typing.Literal[-1, 1]
```

And then i would need to run black